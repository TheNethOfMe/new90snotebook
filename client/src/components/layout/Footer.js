import React from "react";

export default () => {
  return (
    <div>
      <footer>Copyright &copy; {new Date().getFullYear()} Neth Horn</footer>
    </div>
  );
};
