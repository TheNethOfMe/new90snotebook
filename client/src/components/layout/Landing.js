import React, { Component } from "react";
import { Link } from "react-router-dom";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";

export class Landing extends Component {
  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push("/dashboard");
    }
  }
  render() {
    return (
      <div>
        <div className="landing__notebook">
          <div className="landing__subheading">
            Planner, Notebook, and More with 90's Flair!
          </div>
          <ul>
            <li>
              <Link className="landing__nav-item" to="/login">
                <p>Login</p>
              </Link>
            </li>
            <li>
              <Link className="landing__nav-item" to="/register">
                <p>Register</p>
              </Link>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

Landing.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Landing);
