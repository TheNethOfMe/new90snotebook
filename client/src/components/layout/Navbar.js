import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../../actions/authActions";
import { clearCurrentProfile } from "../../actions/profileActions";
import Menu from "../common/Menu";

export class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuToggle: false
    };
  }
  onLogoutClick = e => {
    e.preventDefault();
    this.props.clearCurrentProfile();
    this.props.logoutUser();
  };
  toggleMenu = () => {
    this.setState({ menuToggle: !this.state.menuToggle });
  };
  returnIcon = icon => {
    return `/images/themes/${this.props.theme}/${icon}-icon.png`;
  };
  render() {
    const { isAuthenticated } = this.props.auth;
    const showLinks = isAuthenticated && this.state.menuToggle;
    const authLinks = (
      <ul className="header__nav-list">
        <li>
          <div className="header__nav-item" onClick={this.toggleMenu}>
            <img
              className="header__nav-icon"
              src={this.returnIcon("menu")}
              alt="menu-icon"
            />
            <p>Menu</p>
          </div>
        </li>
        <li>
          <Link className="header__nav-item" to="/dashboard">
            <img
              className="header__nav-icon"
              src={this.returnIcon("settings")}
              alt="settings-icon"
            />
            <p>Settings</p>
          </Link>
        </li>
        <li>
          <div className="header__nav-item" onClick={this.onLogoutClick}>
            <img
              className="header__nav-icon"
              src={this.returnIcon("logout")}
              alt="logout-icon"
            />
            <p>Logout</p>
          </div>
        </li>
      </ul>
    );

    const guestLinks = (
      <ul className="header__nav-list">
        <li>
          <Link className="header__nav-item" to="/login">
            <p>Login</p>
          </Link>
        </li>
        <li>
          <Link className="header__nav-item" to="/register">
            <p>Register</p>
          </Link>
        </li>
      </ul>
    );

    return (
      <div>
        <nav className="header__nav">
          <div className="header__nav-section">
            <ul>
              <li>
                <Link className="header__nav-item" to="/">
                  <p>My 90s Notebook</p>
                </Link>
              </li>
            </ul>
          </div>
          <div className="header__nav-section">
            {isAuthenticated ? authLinks : guestLinks}
          </div>
        </nav>
        {showLinks && <Menu />}
      </div>
    );
  }
}

Navbar.propTypes = {
  auth: PropTypes.object.isRequired,
  theme: PropTypes.string.isRequired,
  logoutUser: PropTypes.func.isRequired,
  clearCurrentProfile: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  theme: state.profile.theme ? state.profile.theme : "paper-cup"
});

export default connect(
  mapStateToProps,
  { logoutUser, clearCurrentProfile }
)(Navbar);
