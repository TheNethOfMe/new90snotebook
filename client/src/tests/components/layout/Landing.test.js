import React from "react";
import { shallow } from "enzyme";
import { Landing } from "../../../components/layout/Landing";

let wrapper, auth, history;

beforeEach(() => {
  history = { push: jest.fn() };
});

test("should render the Login page", () => {
  auth = { user: "test", isAuthenticated: false };
  wrapper = shallow(<Landing auth={auth} history={history} />);
  expect(wrapper).toMatchSnapshot();
});

test("should redirect if user is logged in", () => {
  auth = { user: "test", isAuthenticated: true };
  wrapper = shallow(<Landing auth={auth} history={history} />);
  expect(history.push).toHaveBeenCalledWith("/dashboard");
});
