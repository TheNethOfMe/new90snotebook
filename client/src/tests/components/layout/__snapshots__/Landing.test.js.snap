// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    auth={
      Object {
        "isAuthenticated": false,
        "user": "test",
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="landing__notebook"
      >
        <div
          className="landing__subheading"
        >
          Planner, Notebook, and More with 90's Flair!
        </div>
        <ul>
          <li>
            <Link
              className="landing__nav-item"
              replace={false}
              to="/login"
            >
              <p>
                Login
              </p>
            </Link>
          </li>
          <li>
            <Link
              className="landing__nav-item"
              replace={false}
              to="/register"
            >
              <p>
                Register
              </p>
            </Link>
          </li>
        </ul>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="landing__subheading"
          >
            Planner, Notebook, and More with 90's Flair!
          </div>,
          <ul>
            <li>
              <Link
                className="landing__nav-item"
                replace={false}
                to="/login"
              >
                <p>
                  Login
                </p>
              </Link>
            </li>
            <li>
              <Link
                className="landing__nav-item"
                replace={false}
                to="/register"
              >
                <p>
                  Register
                </p>
              </Link>
            </li>
          </ul>,
        ],
        "className": "landing__notebook",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planner, Notebook, and More with 90's Flair!",
            "className": "landing__subheading",
          },
          "ref": null,
          "rendered": "Planner, Notebook, and More with 90's Flair!",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/login"
                >
                  <p>
                    Login
                  </p>
                </Link>
              </li>,
              <li>
                <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/register"
                >
                  <p>
                    Register
                  </p>
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/login"
                >
                  <p>
                    Login
                  </p>
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <p>
                    Login
                  </p>,
                  "className": "landing__nav-item",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "p",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/register"
                >
                  <p>
                    Register
                  </p>
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <p>
                    Register
                  </p>,
                  "className": "landing__nav-item",
                  "replace": false,
                  "to": "/register",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Register",
                  },
                  "ref": null,
                  "rendered": "Register",
                  "type": "p",
                },
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="landing__notebook"
        >
          <div
            className="landing__subheading"
          >
            Planner, Notebook, and More with 90's Flair!
          </div>
          <ul>
            <li>
              <Link
                className="landing__nav-item"
                replace={false}
                to="/login"
              >
                <p>
                  Login
                </p>
              </Link>
            </li>
            <li>
              <Link
                className="landing__nav-item"
                replace={false}
                to="/register"
              >
                <p>
                  Register
                </p>
              </Link>
            </li>
          </ul>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="landing__subheading"
            >
              Planner, Notebook, and More with 90's Flair!
            </div>,
            <ul>
              <li>
                <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/login"
                >
                  <p>
                    Login
                  </p>
                </Link>
              </li>
              <li>
                <Link
                  className="landing__nav-item"
                  replace={false}
                  to="/register"
                >
                  <p>
                    Register
                  </p>
                </Link>
              </li>
            </ul>,
          ],
          "className": "landing__notebook",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planner, Notebook, and More with 90's Flair!",
              "className": "landing__subheading",
            },
            "ref": null,
            "rendered": "Planner, Notebook, and More with 90's Flair!",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    className="landing__nav-item"
                    replace={false}
                    to="/login"
                  >
                    <p>
                      Login
                    </p>
                  </Link>
                </li>,
                <li>
                  <Link
                    className="landing__nav-item"
                    replace={false}
                    to="/register"
                  >
                    <p>
                      Register
                    </p>
                  </Link>
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="landing__nav-item"
                    replace={false}
                    to="/login"
                  >
                    <p>
                      Login
                    </p>
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <p>
                      Login
                    </p>,
                    "className": "landing__nav-item",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Login",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": "p",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="landing__nav-item"
                    replace={false}
                    to="/register"
                  >
                    <p>
                      Register
                    </p>
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <p>
                      Register
                    </p>,
                    "className": "landing__nav-item",
                    "replace": false,
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Register",
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": "p",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
